addons/repair/models/repair.py

```diff
@@ -130,6 +130,16 @@ class Repair(models.Model):
     invoice_state = fields.Selection(string='Invoice State', related='invoice_id.state')
     priority = fields.Selection([('0', 'Normal'), ('1', 'Urgent')], default='0', string="Priority", help="Important repair order")
 
+
+    @api.depends('location_id')
+    def _compute_valid_picking_ids(self):
+        '''
+            computes the domain of picking_id which will be pickings of type return
+        '''
+        for rec in self:
+            return_picking_types = self.env['stock.picking.type'].search([('code', '=', 'outgoing')]).mapped('return_picking_type_id')
```

Comment:
Please use a search_read instead and put the two searches outside the loop for performance reason.
Thanks